src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs

```diff
@@ -279,14 +271,8 @@ private bool ParameterExistsInAllRoutes(ActionModel actionModel, string paramete
 
         private ModelMetadata GetParameterMetadata(ParameterModel parameter)
```

Comment:
Didn't bother inlining this because @rynowak just refactored the code here 😃

Reply:
@rynowak, I'm considering pushing your refactoring further: Rename the method to `IsComplexTypeParameter` and return `metadata.IsComplexType`. That would help catch any new code that actually needs `ModelMetadataKind.Parameter` metadata. **Thoughts?**